main.go

```diff
@@ -134,39 +137,67 @@ func waitForNotification(
 	}
 }
 
-func produceMessages(p stream.Producer, events []*eventqueue.Event, eq *eventqueue.Queue) {
+func produceMessages(p Producer, events []*eventqueue.Event, eq *eventqueue.Queue) {
 	for _, event := range events {
 		msg, err := json.Marshal(event)
 		if err != nil {
 			logger.L.Fatal("Error parsing event", zap.Error(err))
 		}
 
-		p.Messages() <- &stream.Message{
+		topic := topicName(event.TableName)
+		message := &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &topic,
+				Partition: kafka.PartitionAny, // nolint: gotype
+			},
 			Value:     msg,
-			Topic:     topicName(event.TableName),
 			Key:       event.ExternalID,
 			Timestamp: event.CreatedAt,
 		}
-
+		if os.Getenv("DRY_RUN") != "" {
+			logger.L.Info("Would produce message", zap.Any("message", message))
+		} else {
+			deliveryChan := make(chan kafka.Event)
```

Comment:
I wouldn't create a new deliveryChan for every message, but define the deliveryChan before the for loop so it can be reused.

Reply:
hmm, good point.